Daemonset panic after adding policy.cilium.io/proxy-visibility to some pod

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I follow steps https://docs.cilium.io/en/stable/observability/visibility/#proxy-visibility to enable http visibility to for one of my pod.

When I try to access the pod via other pod, the cilium daemonset error with `panic: runtime error: invalid memory address or nil pointer dereference`

### Cilium Version

Client: 1.14.0 b5013e15 2023-07-26T10:28:38+02:00 go version go1.20.5 linux/amd64
Daemon: 1.14.0 b5013e15 2023-07-26T10:28:38+02:00 go version go1.20.5 linux/amd64

### Kernel Version

Linux home 6.4.7-273-tkg-cfs #1 SMP PREEMPT_DYNAMIC TKG Sat, 29 Jul 2023 14:13:47 +0000 x86_64 GNU/Linux


### Kubernetes Version

Server Version: v1.26.7+k3s1


### Sysdump

[cilium-sysdump-20230811-051659.zip](https://github.com/cilium/cilium/files/12317524/cilium-sysdump-20230811-051659.zip)


### Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2553e3d]

goroutine 1950 [running]:
github.com/cilium/cilium/pkg/proxy/logger.NewLogRecord({0x3605878, 0x7}, 0x0?, {0xc0009f5a18, 0x4, 0xc001010100?})
    /go/src/github.com/cilium/cilium/pkg/proxy/logger/logger.go:95 +0x23d
github.com/cilium/cilium/pkg/envoy.logRecord(0xc002b958c0)
    /go/src/github.com/cilium/cilium/pkg/envoy/accesslog_server.go:209 +0x89e
github.com/cilium/cilium/pkg/envoy.(*AccessLogServer).handleConn(0xc000fca0e0, {0x3bb4138?, 0xc0002f7040}, 0xc0011c02d8)
    /go/src/github.com/cilium/cilium/pkg/envoy/accesslog_server.go:139 +0x518
created by github.com/cilium/cilium/pkg/envoy.StartAccessLogServer.func1
    /go/src/github.com/cilium/cilium/pkg/envoy/accesslog_server.go:82 +0x138
```


### Anything else?

I think it's because [l7tags](https://github.com/cilium/cilium/blob/b5013e1562dbfe442dd2011a5af46de9095315ff/pkg/envoy/accesslog_server.go#L160C14-L160C14) is nil. And [tagFn](https://github.com/cilium/cilium/blob/b5013e1562dbfe442dd2011a5af46de9095315ff/pkg/proxy/logger/logger.go#L95) can not be called

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/pkg/envoy/accesslog_server.go
+++ b/pkg/envoy/accesslog_server.go
@@ -170,7 +170,8 @@ func logRecord(pblog *cilium.LogEntry) *logger.LogRecord {
 	var kafkaRecord *accesslog.LogRecordKafka
 	var kafkaTopics []string
 
-	var l7tags logger.LogTag
+	var l7tags logger.LogTag = func(lr *logger.LogRecord) {}
+
 	if httpLogEntry := pblog.GetHttp(); httpLogEntry != nil {
 		l7tags = logger.LogTags.HTTP(&accesslog.LogRecordHTTP{
 			Method:          httpLogEntry.Method,
@@ -222,7 +223,9 @@ func logRecord(pblog *cilium.LogEntry) *logger.LogRecord {
 	r := logger.NewLogRecord(flowType, pblog.IsIngress,
 		logger.LogTags.Timestamp(time.Unix(int64(pblog.Timestamp/1000000000), int64(pblog.Timestamp%1000000000))),
 		logger.LogTags.Verdict(GetVerdict(pblog), pblog.CiliumRuleRef),
-		logger.LogTags.Addressing(addrInfo), l7tags)
+		logger.LogTags.Addressing(addrInfo),
+		l7tags,
+	)
 	r.Log()
 
 	// Each kafka topic needs to be logged separately, log the rest if any

